### PR TITLE
update to release.openshift.io/feature-set to match OCP 4.12

### PR DESCRIPTION
**Pull request by deads2k**
This was added in
https://github.com/openshift/cluster-version-operator/pull/821 to allow
more featuresets and allow for a future migration to include actual gates

/cc @wking

Looking at this repo, it was not present in 4.11, so there is no migration concern.

### CODE REVIEW

**Review by exdx — APPROVED** [review]
/lgtm

**Comment by exdx** [comment]
Hi @deads2k, the verify check is failing because the new annotation is not being applied to the PO CRD. We would need to update it here https://github.com/openshift/platform-operators/blob/b205d77044f9427ec136e7d6da1da65f4cff8183/Makefile#L82 or apply a patch to the configs directory.

**Comment by deads2k** [comment]
> Hi @deads2k, the verify check is failing because the new annotation is not being applied to the PO CRD. We would need to update it here

Looks like the CRD manifest is copied from the vendor folder (this is fine).  Not sure what the additional annotation is about, but I have fixed openshift/api and revendored to see if that fixes it

File: manifests/0000_50_cluster-platform-operator-manager_07-aggregated-clusteroperator.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ metadata:
   annotations:
     include.release.openshift.io/ibm-cloud-managed: "true"
     include.release.openshift.io/self-managed-high-availability: "true"
-    release.openshift.io/feature-gate: TechPreviewNoUpgrade
+    release.openshift.io/feature-set: TechPreviewNoUpgrade
   name: platform-operators-aggregated
   namespace: openshift-platform-operators
```
Comment (wking) [inline manifests/0000_50_cluster-platform-operator-manager_07-aggregated-clusteroperator.yaml:9]: ClusterOperator [are cluster-scoped resources][1], so this line is not necessary.

[1]: https://github.com/openshift/api/blob/cce310ad2932f6de24491052d506926e484c082c/config/v1/types_cluster_operator.go#L9

**Comment by wking** [comment]
[e2e-techpreview][1]:

```
  Error: Unexpected non-nil/non-zero argument at index 1:
  	<*errors.StatusError>: &errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"clusteroperators.config.openshift.io \"platform-operators-aggregated\" not found", Reason:"NotFound", Details:(*v1.StatusDetails)(0xc00099a6c0), Code:404}}
  In [It] at: /go/src/github.com/openshift/platform-operators/test/e2e/aggregated_clusteroperator_test.go:38
```

Looking at CVO logs, I'm unclear about why.  The CVO knows it's `TechPreviewNoUpgrade`, but has nothing to say about `platform-operators-aggregated`:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_platform-operators/45/pull-ci-openshift-platform-operators-main-e2e-techpreview/1567966053030432768/artifacts/e2e-techpreview/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-5cb8b76556-jv6mw_cluster-version-operator.log | grep 'featurechangestopper\|platform-operators-aggregated'
I0908 20:32:24.570421       1 featurechangestopper.go:100] Starting stop-on-featureset-change controller with "TechPreviewNoUpgrade".
```

Or, I guess, anything in `openshift-platform-operators` either.  Perhaps [the `io.openshift.release.operator` label][2] isn't set yet, and this presubmit is currently a dud?

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_platform-operators/45/pull-ci-openshift-platform-operators-main-e2e-techpreview/1567966053030432768#1:build-log.txt%3A75-77
[2]: https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#product-builds-and-becoming-part-of-an-openshift-release-image

**Comment by timflannagan** [comment]
@wking Yeah, the io.openshift.release.operator Dockerfile label still needs to be toggled on to true in this repository. That's being done in #23 but in the meantime, there won't be any e2e feedback. We were seeing e2e runs in that 23 PR fail because the aggregate platform operators CO resource didn't exist, but looking at a recent run it looks like we're back in business.

**Review by timflannagan — APPROVED** [review]

**Comment by timflannagan** [comment]
These changes are already present in the #23 which has passed the e2e suite in recent runs. We're still in the process of bootstrapping this repository while we wait for payload introduction, so I'm going to add the requisite no-ff labels until then.

/label px-approved
/label docs-approved
/label qe-approved

**Pull request merged**